### ISSUE TITLE
Trigger "change" Events of Input Element

### ISSUE THREAD

**Issue by hkvstore** [Eonasdan/tempus-dominus#2401]
For example, if we use the [Simple Setup (jQuery)](https://getdatepicker.com/6/examples/jquery.html#simple-setup) where the input element has an input group and has a jQuery event attached by:
```js
$("#datetimepicker1Input").on("change", () => alert('something'));
```
then we convert the input to datetime picker by:
```js
new tempusDominus.TempusDominus(document.getElementById('datetimepicker1'));
```

The original "change" event of the input will not be fired unless the original event is attached to the ``div#datetimepicker1`` and the event name is "change.td".  

It would be more convenient if the datetime picker can trigger the input's original events. This can be done by:
1. Change the event type from "change.td" to "change" before triggering the original events (either jQuery event or native HTML element event)
2. Fire the event attached to the input element (not the enclosing DIV)

**Comment by Eonasdan** [comment]
The change event is on w/e the target is, which is this case is the div element. If you have an input field doesn't the native change event take place on that as well? v6 does not require an input field to work so I make very little assumptions on if an input field is in use or not.

**Comment by hkvstore** [comment]
A simple example: (Same as [Simple Setup](https://getdatepicker.com/6/examples/jquery.html#simple-setup))
```html
          <div class="input-group log-event" id="datetimepicker1" data-td-target-input="nearest" data-td-target-toggle="nearest">
            <input id="datetimepicker1Input" type="text" class="form-control" data-td-target="#datetimepicker1">
            <span class="input-group-text" data-td-target="#datetimepicker1" data-td-toggle="datetimepicker">
               <span class="fas fa-calendar"></span>
             </span>
          </div>
```
```js
// 1. Before adding datetime picker, normally event will be attached to the <input> element
// Now if you enter a date manually, leave the field, the following event works
$('#datetimepicker1Input').on('change', () => console.log('change event from #datetimepicker1Input'));

// 2. Add datetime picker, select a date, leave the field, above event does not work any more
$('#datetimepicker1').tempusDominus();

// 3. Attach another event to the enclosing <div> element and change the event type form 'change' to the namespaced 'change.td', it works because it is trigger by the datetime picker.
// Now if enter a date manually, leave the field, BOTH event will be fired
$('#datetimepicker1').on('change.td', () => console.log('change.td event from #datetimepicker'));

// So a better way may be, the datetime picker should automatically trigger the original event in step 1 so that the step 3 is not required, as a result, only one event will be fired.
```

**Comment by Eonasdan** [comment]
Ok I see. Let me play with it and see what I can do.

The outcome you'd like to see is that if the picker is attached to an input field and a user changes the input value that there should be a plain old `change` event on the field e.g. step 1 right?

**Comment by iacobfred** [comment]
Hi @Eonasdan, any updates on this? It would be nice to be able to use the plain old `change` event when using the datetime picker on an input element.

**Comment by Eonasdan** [comment]
Hello, please check the dev branch. You can test out the change on [this stackblitz](https://stackblitz.com/edit/tempus-dominus-v6-next?file=index.js)

**Comment by hkvstore** [comment]
It does not work as expected. e.g.
```js
document
  .getElementById('datetimepicker1Input')
  .addEventListener('change', () => alert('change event fired'));

new tempusDominus.TempusDominus(document.getElementById('datetimepicker1'), {
  //put your config here
});
```
The change event is only fired when the input loses focus, not fired after the datetime picker changes the data and close the calendar.

**Comment by Eonasdan** [comment]
The events are namespaced. You're looking for `change.td`

**Comment by hkvstore** [comment]
But the point of this issue was to trigger the ORIGINAL (non namespaced) " events (e.g. "change") already binded to the input element (before ``tempusDominus.TempusDominus`` is created) so users do not need to add handlers for the namespaced ("change.td") event. (Please refer to the first post.)

**Comment by Eonasdan** [comment]
The namespaced events were a specific request in back in v3. Specially for jQuery. I haven't used jquery but I thought `on` was suppose to bubble namespaced events.

I could remove the `.td` from all event types but this would be a breaking change.

@William-H-M @jcompagner do either of you have an opinion on this?

**Comment by jcompagner** [comment]
why the "change" event is not being fired is because if you set it though the picker. you set it to the value through code and then the change even is not done (i would personally really like that the datepicker would obey to that same rule, but thats another story ;) )

But what is the Event object .... if we rename it (i am fine with that) then the event object of "change" is still the ChangeEvent of the datepicker itself right if you change it through the picker? not the dom change event...
And if you change it by typing in the input field you would get a different kind of event object...

And that is a bit weird....

So i would say keep the current "change.td" (that fires when changing the input or through a picker)
but also maybe just fire also a "change" event but then with a dom event? (CustomEvent object which mimics change event?)

**Comment by Eonasdan** [comment]
@jcompagner I recently added code to the dev branch that triggers the change event on the input as well (if one is present).

The question is mostly is it still a good practice to namespace these events? Bootstrap namespaces their events but they don't have an input manipulate like this.

**Comment by William-H-M** [comment]
Personally, I like the namespaced events as it is easier to detect who is doing what, but some other projects like [bootstrap-selectpikcer](https://developer.snapappointments.com/bootstrap-select/) do use the default events and [namspaced events](https://developer.snapappointments.com/bootstrap-select/options/#events)

To avoid breaking things, why not keep both?

**Comment by jchafik** [comment]
I've added a PR which triggers a native change event on an underlying input when available.

When an input element is updated, a native change event should be triggered. This allows client code to exist without needing explicit knowledge of the date picker used under the hood. Many change events (validation, chaining updates from one input to another, etc.) can and should be agnostic to the plugins (if any) that drive them.

The default namespaced events are retained to avoid a breaking change as well as allow plugin specific listeners.

https://github.com/Eonasdan/tempus-dominus/pull/2533

**Comment by scoldwell** [comment]
@Eonasdan any chance @jchafik's PR for this issue can be approved and merged and a new release with this and the fix for #2527 can be pushed out?  Thanks!

**Comment by scoldwell** [comment]
@Eonasdan I see you merged the PR.  Any update on a release?  Thanks!

**Comment by Eonasdan** [comment]
Just published 5.1. Please check that out
